src/fortran_routing/mc_pylink_v00/Reservoir_singleTS/module_levelpool.f90

```diff
@@ -0,0 +1,237 @@
+! This module derived from the NWM repo found here: 
+! https://github.com/NCAR/wrf_hydro_nwm_public/blob
+! /master/trunk/NDHMS/Routing/Reservoirs/Level_Pool/module_levelpool.F
+
+! This module defines and instantiates objects
+! for a level pool type reservoir. The level
+! pool reservoir type inherits input and
+! output types from the reservoir base
+! module and calls instantiation of these into
+! sub-objects. The level pool reservoir type
+! also points to types for level pool properties
+! and state and calls instantiation of these into
+! sub-objects. This module also contains the
+! subroutine to run level pool reservoir that is
+! derived from the reservoir base type interface
+! to run reservoir. Running level pool will
+! then call the LEVELPOOL_PHYSICS subroutine, which
+! processes the given inputs, properties, and
+! state for a particular level pool reservoir and
+! returns the output/outflow.
+
+module module_levelpool
```

Comment:
@kmsampson
Kevin, 
I'm tagging you on the PR into our routing repository... hopefully it's not too confusing. I thought you might be interested. We used the f2py wrapper to build a python call directly to the level pool function from the WRF-Hydro code. We've made a few minor modifications such as specifying the precision (because f2py is not careful with that in the translation) and making the depth variable more explicit (in - H0, out - H1). Eventually, we will use a custom cython wrapper so we can operate outside the GIL.